docs/adr/adr.template

```diff
@@ -0,0 +1,29 @@
+# ADR template by Michael Nygard
+
+This template is adapted from [Documenting architecture decisions template by Michael Nygard](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md).
```

Comment:
I like the link to the blog post with guidance on how best to fill this out.  This'll be helpful for me when I come to do one :-)

Reply:
I have added the link back into the original blog post and have updated the links section as well, renaming to `Supporting Documentation` so that it is more generic, i.e: links to Trello cards and SharePoint resources